File: 퇴사.py
def go(today,profit,lastDay,t,p):
  if today > lastDay: return 0
  if today == lastDay: return profit
  ret = max(go(today+1,profit,lastDay,t,p),go(today+t[today],profit+p[today],lastDay,t,p))
  return ret
n= int(input())
t= [0]*n
p=[0]*n
for i in range(n):
  t[i],p[i]= map(int,input().split())
print(go(0,0,n,t,p))

File: 퇴사.cpp
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int go(int today,int profit,int lastDay,vector<int>& t,vector<int>& p){
  if(today > lastDay) return 0;
  if(today == lastDay) return profit;
  int ret = go(today+1,profit,lastDay,t,p);
  ret= max(ret,go(today+t[today],profit+p[today],lastDay,t,p));
  return ret;
}

int main(){
  int n; 
  cin>>n;
  vector<int> t(n);
  vector<int> p(n);
  for(int i=0; i<n; i++){
    cin>>t[i]>>p[i];
  }
  cout<<go(0,0,n,t,p)<<'\n';
}